WooCommerce/Classes/ViewRelated/Products/Variations/Add Attributes/AddAttributeOptionsViewModel.swift

```diff
@@ -53,20 +53,14 @@ final class AddAttributeOptionsViewModel {
 
     /// Name of the attribute
     ///
-    var attributeName: String {
-        switch attribute {
-        case .new(let name):
-            return name
-        case .existing(let attribute):
-            return attribute.name
-        }
-    }
+    private(set) var currentAttributeName: String
```

Comment:
Since this is now part of the state of the mutable state of the view model, I think you can move it inside the state struct. This would allow us to remove the `self?.renderViewModel()` that you added in the `navigateToRenameAttribute` method to make sure that the `VC` is always rendering the latest `VM` state.

Reply:
Oh, good point, thanks! I made that change in https://github.com/woocommerce/woocommerce-ios/pull/3781/commits/d061bcbc4383f0eee553c0db753c93a52fa312ff. The changes work, but let me know if there are better conventions for that (especially things like initializing `currentAttributeName` and the setter/getter methods since the state struct is entirely private).